lib/facility.rb

```diff
@@ -1,14 +1,91 @@
+require 'date'
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
   end
-
-  def add_services(service)
+  
+  def add_service(service)
+    self.collected_fees_helper
+    if service == 'Vehicle Registration'
+      self.register_vehicle_helper
+    end
     @services << service
   end
+
+  def registered_vehicles
+    self.register_vehicle_helper
+    @registered_vehicles
+  end
+
+  def collected_fees
+    self.collected_fees_helper
+    @collected_fees
+  end
+
+  def register_vehicle_helper
+    if @registered_vehicles.nil?
+      @registered_vehicles = []
+    end
+  end
```

Comment:
I would use caution declaring/assigning values to ivars outside of the `initialize` method. It can cause issues with clarity and code readability down the road as projects scale. 
In my opinion it would be fine to instantiate every `Facility` with `@registered_vehicles = []` so that you wouldn't need this method.